### PR TITLE
#1093: Update stock quantity bug

### LINKED ISSUE #1093: Update stock quantity bug
When the product quantity in stock is reduced, the value by which we want to reduce the quantity can be greater than the current quantity. In this case the product quantity is negative

### PR DESCRIPTION
**Pull request by s04v**
Fix #1093

### CODE REVIEW

File: src/Modules/SimplCommerce.Module.Inventory/Services/StockService.cs [hunk 1]
```diff
@@ -48,9 +48,16 @@ public async Task UpdateStock(StockUpdateRequest stockUpdateRequest)
             var product = await _productRepository.Query().FirstOrDefaultAsync(x => x.Id == stockUpdateRequest.ProductId);
             var stock = await _stockRepository.Query().FirstOrDefaultAsync(x => x.ProductId == stockUpdateRequest.ProductId && x.WarehouseId == stockUpdateRequest.WarehouseId);
 
+            var adjustedQuantity = stockUpdateRequest.AdjustedQuantity;
+
+            if (adjustedQuantity < 0 && Math.Abs(adjustedQuantity) > stock.Quantity)
```
Comment (hishamco) [inline src/Modules/SimplCommerce.Module.Inventory/Services/StockService.cs:53]:
```suggestion
            var adjustedQuantity = stockUpdateRequest.AdjustedQuantity;
            if (adjustedQuantity < 0 && Math.Abs(adjustedQuantity) > stock.Quantity)
```

**Comment by hishamco** [comment]
A unit test will be useful to ensure nothing broken in this case

**Comment by s04v** [comment]
Hello, @hishamco, there is a fully commented out [StockServiceTests.cs](https://github.com/simplcommerce/SimplCommerce/blob/master/test/SimplCommerce.Module.Inventory.Tests/StockServiceTests.cs) file . Should I create a new file for the tests?

**Comment by hishamco** [comment]
Seems it's commented :) you can uncomment the class and add your test cases, leave everything as it was

**Comment by hishamco** [comment]
Why you are using `MockQueryable.Moq`

**Comment by s04v** [comment]
Because the `FirstOrDefaultAsync` method is used in `UpdateStock`. Calling `FirstOrDefaultAsync` with Moq throws an exception that this method works only with IAsyncQueryProvider. MockQueryable.Moq helps to solve it

**Comment by hishamco** [comment]
I see, let me do a quick check today then merge

Thanks

**Comment by s04v** [comment]
Any updates?

**Comment by hishamco** [comment]
Sorry I was sick last few days, that's why I didn't give you an update

**Comment by s04v** [comment]
It's okay, take care of yourself.

**Pull request merged**